quic/s2n-quic-events/events/common.rs

```diff
@@ -40,6 +40,7 @@ struct TransportParameters<'a> {
     initial_max_streams_bidi: u64,
     initial_max_streams_uni: u64,
     max_datagram_frame_size: u64,
+    dc_supported_versions: [u32; 4],
```

Comment:
Can we make this a `&'a [u32]`? I'd prefer the events stays consistent no matter what we do in the future.

Reply:
Hmm, this is giving me trouble. Is there a way to do this without doing some allocation?

I've been trying something like this
```rust
impl<'a> IntoEvent<&'a [u32]> for DcSupportedVersions {
    fn into_event(self) -> &'a [u32] {
        self.versions
            .map(|version| version.as_u64() as u32)
            .as_slice()
    }
}
```
and it gives me `cannot return reference to temporary value`

Reply:
You need to implement `IntoEvent` for a reference, similar to https://github.com/aws/s2n-quic/blob/1436af712b6e73edc11640dc7c3cae23e456c0a8/quic/s2n-quic-events/events/common.rs#L52-L64